Source/WebKit/WebProcess/Network/WebSocketChannel.cpp

```diff
@@ -147,16 +148,15 @@ WebSocketChannel::ConnectStatus WebSocketChannel::connect(const URL& url, const
     std::optional<FrameIdentifier> frameID;
     std::optional<PageIdentifier> pageID;
     StoredCredentialsPolicy storedCredentialsPolicy { StoredCredentialsPolicy::Use };
-    if (auto* frame = m_document ? m_document->frame() : nullptr) {
-        RefPtr mainFrame = m_document->localMainFrame();
+    if (RefPtr frame = document ? document->frame() : nullptr) {
```

Comment:
The null check here doesn't make sense since you've already done it at the beginning on this function and early returned when null. Previously, we were checking a WeakPtr so it could in theory become null after calling `client->didUpgradeURL()` above for example. However, this is no longer possible after your code change.